Bottom: dcbfb2c35318794c22c1c7bc76753c2213d56e01
Top:    514ea88f8631add7f29fd4864109d06c586ef744
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 23:24:11 +0100

Refresh of add-a-hid-api-call-notify_save

---

diff --git a/src/file.c b/src/file.c
index 3f97364..7c1ebc0 100644
--- a/src/file.c
+++ b/src/file.c
@@ -355,7 +355,7 @@ SavePCB (char *Filename)
   char *copy;
 
   if (gui->notify_save_pcb != NULL)
-    gui->notify_save_pcb (false);
+    gui->notify_save_pcb (Filename, false);
 
   if (!(retcode = WritePipe (Filename, true)))
     {
@@ -370,7 +370,7 @@ SavePCB (char *Filename)
     }
 
   if (gui->notify_save_pcb != NULL)
-    gui->notify_save_pcb (true);
+    gui->notify_save_pcb (Filename, true);
 
   return (retcode);
 }
diff --git a/src/hid.h b/src/hid.h
index afb33e7..d1c7c8e 100644
--- a/src/hid.h
+++ b/src/hid.h
@@ -576,7 +576,7 @@ typedef enum
      * Allows GUIs which watch for file-changes on disk to ignore
      * our deliberate changes.
      */
-    void (*notify_save_pcb) (bool done);
+    void (*notify_save_pcb) (const char *filename, bool done);
   };
 
 /* Call this as soon as possible from main().  No other HID calls are
